src/routes/GroupSettings/GroupSettings.js

```diff
@@ -49,103 +53,139 @@ export default function GroupSettings ({
     group && fetchGroupSettings()
   }, [slug])
 
+  const responsibilities = getResponsibilitiesForGroup({ currentUser, groupId: group?.id }).map(r => r.title)
   if (!group) return <Loading />
-  if (!canModerate) return <Redirect to={groupUrl(slug)} />
+  if (!canModerate && responsibilities.length === 0) return <Redirect to={groupUrl(slug)} />
+
+  useEffect(() => {
+    if (!canModerate && !responsibilities.includes(RESP_ADMINISTRATION) && responsibilities.includes(RESP_ADD_MEMBERS)) push(groupUrl(slug, 'settings/invite'))
+  }, [])
+
+  const overallSettings = {
+    name: t('Settings'),
+    path: groupUrl(slug, 'settings'),
+    component: <GroupSettingsTab
+      fetchLocation={fetchLocation}
+      fetchPending={fetchPending}
+      currentUser={currentUser}
+      group={group}
+      parentGroups={parentGroups}
+      prerequisiteGroups={prerequisiteGroups}
+      updateGroupSettings={updateGroupSettings}
+    />
+  }
+
+  const agreementSettings = {
+    name: t('Agreements'),
+    path: groupUrl(slug, 'settings/agreements'),
+    component: <AgreementsTab group={group} />
+  }
+
+  const responsibilitiesSettings = {
```

Comment:
do you think we can have roles and responsibilities in one tab? that would be nice

Reply:
It'll be huge is all